cmd/chisel/cmd_cut_test.go

```diff
@@ -0,0 +1,560 @@
+			package: test-package
+			slices:
+				myslice:
+					essential:
+						- test-package_manifest
+					contents:
+						/dir/file:
+						/db/**:   {generate: manifest}
+				manifest:
+					contents:
+						/db/**:   {generate: manifest}
+						/db-2/**: {generate: manifest}
+		`,
+	},
+	slices: []string{"test-package_myslice"},
+	filesystem: map[string]string{
+		"/db-2/":          "dir 0755",
+		"/db-2/chisel.db": "file 0644 4a51075c",
+		"/db/":            "dir 0755",
+		"/db/chisel.db":   "file 0644 4a51075c",
+		"/dir/":           "dir 0755",
+		"/dir/file":       "file 0644 cc55e2ec",
+	},
+	db: map[string]string{
+		"/db/chisel.db": strings.TrimLeft(`
+{"jsonwall":"1.0","schema":"1.0","count":11}
+{"kind":"content","slice":"test-package_manifest","path":"/db-2/chisel.db"}
+{"kind":"content","slice":"test-package_manifest","path":"/db/chisel.db"}
+{"kind":"content","slice":"test-package_myslice","path":"/db/chisel.db"}
+{"kind":"content","slice":"test-package_myslice","path":"/dir/file"}
+{"kind":"package","name":"test-package","version":"test-package_version","sha256":"test-package_hash","arch":"test-package_arch"}
+{"kind":"path","path":"/db-2/chisel.db","mode":"0644","slices":["test-package_manifest"]}
+{"kind":"path","path":"/db/chisel.db","mode":"0644","slices":["test-package_manifest","test-package_myslice"]}
+{"kind":"path","path":"/dir/file","mode":"0644","slices":["test-package_myslice"],"sha256":"cc55e2ecf36e40171ded57167c38e1025c99dc8f8bcdd6422368385a977ae1fe","size":14}
+{"kind":"slice","name":"test-package_manifest"}
+{"kind":"slice","name":"test-package_myslice"}
+`, "\n"),
+		"/db-2/chisel.db": strings.TrimLeft(`
+{"jsonwall":"1.0","schema":"1.0","count":11}
+{"kind":"content","slice":"test-package_manifest","path":"/db-2/chisel.db"}
+{"kind":"content","slice":"test-package_manifest","path":"/db/chisel.db"}
+{"kind":"content","slice":"test-package_myslice","path":"/db/chisel.db"}
+{"kind":"content","slice":"test-package_myslice","path":"/dir/file"}
+{"kind":"package","name":"test-package","version":"test-package_version","sha256":"test-package_hash","arch":"test-package_arch"}
+{"kind":"path","path":"/db-2/chisel.db","mode":"0644","slices":["test-package_manifest"]}
+{"kind":"path","path":"/db/chisel.db","mode":"0644","slices":["test-package_manifest","test-package_myslice"]}
+{"kind":"path","path":"/dir/file","mode":"0644","slices":["test-package_myslice"],"sha256":"cc55e2ecf36e40171ded57167c38e1025c99dc8f8bcdd6422368385a977ae1fe","size":14}
+{"kind":"slice","name":"test-package_manifest"}
+{"kind":"slice","name":"test-package_myslice"}
+`, "\n"),
+	},
+}, {
+	summary: "Same file mutated across Multiple packages",
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					essential:
+						- test-package_manifest
+					contents:
+						/dir/file:
+						/foo:       {text: foo, mutable: true}
+					mutate: |
+						content.write("/foo", "test-package")
```

Comment:
If I am not wrong, `content.write` basically "writes" (not "amends") the provided text/bytes to the file. So, here, the file would be
```
test-package
```

Reply:
But the other slice is writing `"other-package"`, right?

Reply:
Yup. The content will depend on the final order of slices I believe.

Reply:
Can you dive deep and see how they are sorted out? I wonder if it is alphabetically or if the essential field plays a role. This might be important because we might want to disallow it. Not a priority but please do it when you have some time.

Reply:
So it's basically [Tarjan's SCC algorithm](https://en.wikipedia.org/wiki/Tarjan's_strongly_connected_components_algorithm). Looking at the implementation, it should produce a deterministic order. The order should be produced based on the `essential` relation first as a topological sorting would do. Since there can be multiple such ordering, care is given to sort the appropriate portions of list.

I just glimsed through the code, but I will make sure to take a deeper look later.